### ISSUE TITLE
Not able to integrate select2 with FUEL-CMS

### ISSUE THREAD

**Issue by SunYunSoft** [daylightstudio/FUEL-CMS#302]
Hi

We recently use this wonderful CMS, however, we met an issue for use custom field. Andy help is appreciated and thanks in advance. 
I have follow the step to add the select2 into fuel cms, however, we do met issue when adding data to select2 plugin, we have tried several way, but it could not work.

And one more question is, with custom field, how could we set the custom field value and get the custom field value when submitted?

		$js = '<script type="text/javascript">
			    $(function(){
			    	$(".field_type_tag").select2({
...
data: data <------------ adding data would lead select2 not functionable
...
});
		    	})

http://forum.getfuelcms.com/discussion/1958/new-to-fuelcms-is-it-possible-to-create-a-multilingual-cms/p1

**Comment by SunYunSoft** [comment]
PS: we are not able register with the forum, so we could only submit it here.

**Comment by daylightstudio** [comment]
I'm not quite sure I understand the question but will give it a shot. The name of the field (e.g. my_field) will get submitted in the post. If there is a field in your database table of "my_field", then you can access the posted information and further manipulate it in a model's on_before_save/on_after_save hooks (http://docs.getfuelcms.com/general/models#hooks). This allows you to save the information where you need to in a format you want.
```
public function on_before_save($values)
{
     $values = parent_on_after_save($values);
     // ... put your custom code here
     return $values;
}
```
To get access to all the posted data, it can be referenced either simply using the $_POST variable, or there is a reference to it on the model as a property of normalized_save_data (see below):
```
public function on_before_save($values)
{
     $values = parent_on_after_save($values);
     $data = $this->normalized_save_data;
     print_r($data);
     exit(); // remove this after testing
     // ... put your custom code here
     return $values;
}
```
For displaying the correct value of the field, this depends on if there is a field in your module's table of "my_field". If so, FUEL will automatically use that my_field value as the value and you can NOT use the "value" parameter in your form_fields method, unless you further manipulate your model's "find_one_array" method (the default method used to retrieve saved data for a record to edit), or you define a separate "edit_method" for your module (http://docs.getfuelcms.com/modules/simple). 

If "my_field" is not a field in your module's table, then you can specify the "value" parameter and even reference a custom method to retrieve that value:
```
$fields['my_field']['value'] = $this->custom_method_to_get_my_field_value();
```

Thanks for letting me know about the Forum, I will need to look into what the issue is there.

**Comment by daylightstudio** [comment]
You may want to check your SPAM folder for the confirmation emai

**Comment by daylightstudio** [comment]
I don't know if this is related, but I'm noticing the Select2 isn't using the original values from the options list but instead is using its label as the values. Is this what you are seeing too?

**Comment by SunYunSoft** [comment]
Thanks for reply.
1. for using select2, we could not find the my_field value in the $_POST super global, so we need seek here for help.
2. for populate the options list for select 2, we could not set the data property, once we set the data when initialize the select2, the select2 is not working.
    $js = '<script type="text/javascript">
            $(function(){
                $(".field_type_tag").select2({
...
data: data <------------ adding data would lead select2 not functionable
...
});
})
3. for register the forum, we could not register, we always see below notice when signup. However, we don't see a recaptcha value in the register form.
The reCAPTCHA value was not entered correctly. Please try again.

**Comment by daylightstudio** [comment]
Not sure if you know this or not, but you can use an "options" property to populate the options of the select?
```
$fields['my_field'] = array('type' => 'select2', 'options' => array('opt1' => 'Option 1', 'opt2' => 'Option 2')); 
```

Regarding signup, are you saying you don't see the recaptcha image at all?

**Comment by SunYunSoft** [comment]
1) for the options, we need some data set dynamically based on some other selection, so could not set it when form_fields(), and would need get it via ajax. Is there any possible work around or other suggestion?

2) for singup, I could not see any recaptcha image. below is a snapshot. I have tried Sarafi on Mac and IE on Windows.
![a](https://cloud.githubusercontent.com/assets/11387759/6570570/d8bc8596-c737-11e4-8e3c-aba270fd8505.png)

**Comment by daylightstudio** [comment]
You should see something like the following image. What browser are you using and do you have Javascript enabled?

![getfuelcms_apply_for_membership](https://cloud.githubusercontent.com/assets/410984/6590918/a43f01b6-c775-11e4-97a3-1285d90e8135.png)

**Comment by SunYunSoft** [comment]
1. for the selection field, I have turned to use the dependent form field in fuel CMS. And it could solve most of our problems. Thanks.
2. for the recaptcha, I have checked the web page, and notice this is using google service, which is not available in China. So seems we are not able to see it.

**Comment by daylightstudio** [comment]
Interesting about it not working in China.
